packages/world/src/systems/BedSystem.sol

```diff
@@ -136,4 +95,45 @@ contract BedSystem is System {
 
     notify(caller, WakeupNotification({ bed: bed, bedCoord: bedCoord }));
   }
+
+  function removeDeadPlayerFromBed(EntityId player, Vec3 dropCoord) public {
+    checkWorldStatus();
+
+    EntityId bed = PlayerStatus._getBedEntityId(player);
+    require(bed.exists(), "Player is not in a bed");
+
+    Vec3 bedCoord = Position._get(bed);
+
+    // TODO: use a different constant?
+    require(bedCoord.inSurroundingCube(dropCoord, MAX_RESPAWN_HALF_WIDTH), "Drop location is too far from bed");
+
+    (EntityId drop, ObjectTypeId objectTypeId) = getOrCreateEntityAt(dropCoord);
+    require(ObjectTypeMetadata._getCanPassThrough(objectTypeId), "Cannot drop items on a non-passable block");
+
+    (EntityId forceField,) = getForceField(bedCoord);
+    EnergyData memory playerData = BedLib.updateSleepingPlayer(forceField, player, bed, bedCoord);
+
+    require(playerData.energy == 0, "Player is not dead");
+
+    PlayerUtils.removePlayerFromBed(player, bed, forceField);
+
+    BedLib.transferInventory(bed, drop);
+    // TODO: Should we safecall the program?
+  }
+}
+
+// To avoid reaching bytecode size limit
+library BedLib {
```

Comment:
just moved these to the end of the file, no changes

Reply:
ty for the comment, makes it easier to review!